### ISSUE TITLE
Bug: 25M Doesn't Scale Properly According to Curve Formula

### ISSUE THREAD

**Issue by cdenq** [azerothcore/mod-autobalance#154]
### Current Behaviour

Currently, Patchwerk from Naxx @ 25M should have 13M hp (13,503,691 specifically, this number differs from wowpedia articles for some reason). With the autobalance module, at default 5 player scaling on a 25M instance, the scaling should be 0.1235*13M = 1667K hp. However, Patchwerk has 6751K hp, which is like 70% of the full power level at 20% of player capacity. The inflection point / scaling of 25M seems wrong.

### Expected Behaviour

25M scaling should follow the default 25M scaling curve found here: https://docs.google.com/spreadsheets/d/100cmKIJIjCZ-ncWd0K9ykO8KUgwFTcwg4h2nfE_UeCc/copy.

### Steps to reproduce the problem

Launch module and log into 25M Naxx with 5 players.

### AC rev. hash/commit

N/A

### Operating system

Windows 11

**Comment by kjack9** [comment]
Can you provide your entire `AutoBalance.conf` file please?

**Comment by kjack9** [comment]
Thanks for following up! I'm having a hard time reproducing this issue. Let me walk you thorough my testing - tell me if I'm missing something.

All testing done on Patchwerk in 25-man normal Naxx. I am adjusting the Difficulty Offset (`.ab setoffset x`) in order to simulate different numbers of players in the instance, but the math should end up being the same as when players are actually there. I copy/pasted your config into my AC install, so it definitely matches yours. 

I'm using a slightly-updated version of [`mod-autobalance`](https://github.com/kjack9/mod-autobalance/tree/dev) that doesn't have any changes that affect the curve math, but you will notice some differences in the `.ab mapstat` and `.ab creaturestat` outputs.

Here's the curve I'm expecting based on your settings:
![patchwerk curve](https://github.com/azerothcore/mod-autobalance/assets/19712078/1741e2ba-bf0d-4be0-b736-b7392f23412f)

If I set the adjusted player count to 25 (difficulty offset of 24 + me), Patchwerk has a health modifier of 1.000 and 13,038,575 health.
![patchwerk-25-player](https://github.com/azerothcore/mod-autobalance/assets/19712078/f3b85e7c-e70c-4439-aa0d-124c94e4658e)

If I set the adjusted player count to 5 (difficulty offset of 4 + me), Patchwerk has a health modifier of 0.1234554... 

![patchwerk-5-player](https://github.com/azerothcore/mod-autobalance/assets/19712078/bd6b557b-63e3-44f6-8f51-b88d779734e3)

`13,038,575 * 0.1234554 = 1,609,682` health, which is exactly what he's getting.

![patchwerk-5-man-info](https://github.com/azerothcore/mod-autobalance/assets/19712078/bfcbcbc2-4517-48ef-8cd3-8c2b8ce264e4)

If you want to test this further, I'd appreciate you updating to [this version of `mod-autobalance`](https://github.com/kjack9/mod-autobalance/tree/dev). You can download the contents of that zip file straight into the `mod-autobalance` folder in your AC install (maybe backup your existing folder just in case). Recompile and restart your server.

Then recreate the situation where Patchwerk is getting the wrong values. Target him, and issue both the `.ab mapstat` and `ab.creaturestat` commands, then screenshot the results and add them to this ticket. That should give us a good starting point to figuring out what's going on.

**Comment by kjack9** [comment]
Flash thought - have you changed any of these settings in your `worldserver.conf`?

```
Rate.Creature.Normal.HP          = 1
Rate.Creature.Elite.Elite.HP     = 1
Rate.Creature.Elite.RARE.HP      = 1
Rate.Creature.Elite.RAREELITE.HP = 1
Rate.Creature.Elite.WORLDBOSS.HP = 1
```

**Comment by kjack9** [comment]
OH! If the max capacity of the dungeon is reporting to AutoBalance as 10, then the scale is going to be done as though there were 10 players, not 25. Yes you specified the inflection settings for this particular dungeon, but as far as AutoBalance knows it's a 10-player raid.

![image](https://github.com/azerothcore/mod-autobalance/assets/19712078/09f89d36-2b3a-4ba0-8206-2a3572c7dd9b)

Oh hey, that number looks familiar...

You told the game it was a 10-player raid, and the game (and AutoBalance) believed you. So it won't scale based on 25-player anymore - because it's not a 25-player raid.

Update your inflection settings to get the value you want when the Dungeon Max Players is set to 10. That should hopefully resolve your issue.

I DO see a bug in this though:

![image](https://github.com/azerothcore/mod-autobalance/assets/19712078/2427fe7b-5ff0-4116-9ed1-67be698fe228)

I need to figure out why that value doesn't match the max number of players its actually scaling based on. Because clearly it's convinced the raid has a max of 10 players now.

**Comment by kjack9** [comment]
I updated `.ab mapstat` to look like this:

![image](https://github.com/azerothcore/mod-autobalance/assets/19712078/69ae8400-498e-43e8-83e6-d2819ef87ef4)

**Comment by kjack9** [comment]
Do let me know if you're able to isolate what's causing the issue - if it's with `mod-autobalance`, I want to get it fixed. That being said, I'm not able to replicate this issue on a fresh install of `mod-autobalance`, so I'm going to mark this one `invalid` for now. Feel free to re-open if you find out more information.

As to including the Instance ID number, I am concerned that would cause confusion for most users - there are already a lot of IDs involved, and `.ab mapstat` is already pretty information-dense as it is. I AM now outputting what instance type you're in (as seen [here](https://user-images.githubusercontent.com/19712078/269824207-69ae8400-498e-43e8-83e6-d2819ef87ef4.png)), so hopefully that differentiates enough for most use cases.

**Comment by kjack9** [comment]
> Hmm, so interestingly, Patchwerk 25M is the only mob that is affected incorrectly. I checked other 25M raid mobs and they all seem to scale correctly!
> 
> I'm thinking whether there is some sort of id exception made for Patchwerk? Another method, do you mind sharing your `.npc id` logs for both Patchwerk in 10M and Patchwerk in 25M? May be onto something here! :)
> 
> @kjack9

Sure thing, I'll get that for you tonight.

**Comment by kjack9** [comment]
Whelp.

![image](https://github.com/azerothcore/mod-autobalance/assets/19712078/7e1aa116-0940-49c3-b82a-3ca1e0222103)

**Comment by kjack9** [comment]
Looks like that default setting has existed for at least 7 years, since this repo was started. WOW!

**Comment by kjack9** [comment]
Re-opening, as this is caused by the default `AutoBalance.ForcedID40` setting that included the creature ID for Patchwerk (16028).

Closed by #155

**Comment by kjack9** [comment]
Closed by #155
